ably/internal/ablyutil/time.go

```diff
@@ -75,21 +72,19 @@ func ContextWithTimeout(ctx context.Context, after TimerFunc, timeout time.Durat
 
 type contextWithCustomErr struct {
 	context.Context
-	err <-chan error
```

Comment:
This was a buffered channel for a reason: reading and writing the error happens on different goroutines, so it needs to be synchronized.

At the very least, if you really don't like the channel, add a sync.Mutex.

Reply:
No, it won't cause, simultaneous read and write. When the timer is fired, context is not canceled,
so `ctx.Done()` is not yet called, once we set the error, then we cancel the context.

Reply:
Discussed offline; decided OK to leave as-is for the purposes of this PR.